sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.common.export;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Base class for all the config builder classes (SimpleSpanProcessor, BatchSpanProcessor, etc).
+ *
+ * @param <T> the type of the config builder
+ */
+public abstract class ConfigBuilder<T> {
+
+  @VisibleForTesting
+  protected enum NamingConvention {
+    DOT {
+      @Override
+      public String normalize(@Nonnull String key) {
+        return key.toLowerCase();
+      }
+    },
+    ENV_VAR {
+      @Override
+      public String normalize(@Nonnull String key) {
+        return key.toLowerCase().replace("_", ".");
+      }
+    };
+
+    public abstract String normalize(@Nonnull String key);
```

Comment:
What's the rationale behind having `normalize` on both the naming convention enum and the `ConfigBuilder` itself?

Reply:
This method is abstract so each `NamingConvention` has to implement its normalization. It is only available inside the enum.

Reply:
Oooh, my bad. I had read it as being *outside* the enum.